apiserver/facades/agent/uniter/uniter.go

```diff
@@ -332,13 +334,27 @@ func (u *UniterAPI) PrivateAddress(ctx context.Context, args params.Entities) (p
 
 // TODO(ericsnow) Factor out the common code amongst the many methods here.
 
-var getZone = func(st *state.State, tag names.Tag) (string, error) {
+var getZone = func(ctx context.Context, st *state.State, machineService MachineService, tag names.Tag) (string, error) {
 	unit, err := st.Unit(tag.Id())
 	if err != nil {
 		return "", errors.Trace(err)
 	}
-	zone, err := unit.AvailabilityZone()
-	return zone, errors.Trace(err)
+	machineID, err := unit.AssignedMachineId()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	machineUUID, err := machineService.GetMachineUUID(ctx, coremachine.Name(machineID))
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	hc, err := machineService.HardwareCharacteristics(ctx, machineUUID)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if hc.AvailabilityZone == nil {
+		return "", errors.Trace(err)
+	}
```

Comment:
If the `AvailabilityZone` is nil, the error will be nil. If that's ok, then we should just do `return "", nil`. If that's not what we want, then we should return a specific error.

Reply:
You're right, fixed in d468ec13d0